compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/syntax/tree/TreeModifier.java

```diff
@@ -3070,14 +3070,23 @@ protected Node transformSyntaxNode(Node node) {
         }
 
         boolean nodeModified = false;
-        STNode[] newSTNodes = new STNode[nodeList.size()];
+        STNode[] newSTNodes = new STNode[nodeList.internalListNode.size()];
+
         for (int index = 0; index < nodeList.size(); index++) {
             T oldNode = nodeList.get(index);
             T newNode = modifyNode(oldNode);
             if (oldNode != newNode) {
                 nodeModified = true;
             }
-            newSTNodes[index] = newNode.internalNode();
+
+            if (nodeList instanceof SeparatedNodeList && index < nodeList.size() - 1) {
```

Comment:
Can we use separate methods for modifying `NodeList` and `SeparatedNodeList`?

Reply:
ok